test_pytest_rerunfailures.py

```diff
@@ -385,57 +457,91 @@ def setup_fixture(self):
 
 
 def test_execution_count_exposed(testdir):
-    testdir.makepyfile('def test_pass(): assert True')
-    testdir.makeconftest("""
+    testdir.makepyfile("def test_pass(): assert True")
+    testdir.makeconftest(
+        """
         def pytest_runtest_teardown(item):
-            assert item.execution_count == 3""")
-    result = testdir.runpytest('--reruns', '2')
+            assert item.execution_count == 3"""
+    )
+    result = testdir.runpytest("--reruns", "2")
     assert_outcomes(result, passed=3, rerun=2)
 
 
 def test_rerun_report(testdir):
-    testdir.makepyfile('def test_pass(): assert False')
-    testdir.makeconftest("""
+    testdir.makepyfile("def test_pass(): assert False")
+    testdir.makeconftest(
+        """
         def pytest_runtest_logreport(report):
             assert hasattr(report, 'rerun')
             assert isinstance(report.rerun, int)
             assert report.rerun <= 2
-        """)
-    result = testdir.runpytest('--reruns', '2')
+        """
+    )
+    result = testdir.runpytest("--reruns", "2")
     assert_outcomes(result, failed=1, rerun=2, passed=0)
 
 
 def test_pytest_runtest_logfinish_is_called(testdir):
     hook_message = "Message from pytest_runtest_logfinish hook"
-    testdir.makepyfile('def test_pass(): pass')
-    testdir.makeconftest(r"""
+    testdir.makepyfile("def test_pass(): pass")
+    testdir.makeconftest(
+        r"""
         def pytest_runtest_logfinish(nodeid, location):
-            print("\n{0}\n")
-    """.format(hook_message))
-    result = testdir.runpytest('--reruns', '1', '-s')
+            print("\n{}\n")
+    """.format(
+            hook_message
+        )
+    )
+    result = testdir.runpytest("--reruns", "1", "-s")
     result.stdout.fnmatch_lines(hook_message)
 
+
 @pytest.mark.parametrize(
-        "file_text, only_rerun_texts, should_rerun",
-        [
-            ('def test_only_rerun(): raise AssertionError("ERR")', ['AssertionError'], True),
-            ('def test_only_rerun(): raise AssertionError("ERR")', ['Assertion*'], True),
-            ('def test_only_rerun(): raise AssertionError("ERR")', ['Assertion'], True),
-            ('def test_only_rerun(): raise AssertionError("ERR")', ['ValueError'], False),
-            ('def test_only_rerun(): raise AssertionError("ERR")', [''], True),
-            ('def test_only_rerun(): raise AssertionError("ERR")', ['AssertionError: '], True),
-            ('def test_only_rerun(): raise AssertionError("ERR")', ['AssertionError: ERR'], True),
-            ('def test_only_rerun(): raise AssertionError("ERR")', ['ERR'], True),
-            ('def test_only_rerun(): raise AssertionError("ERR")', ['AssertionError,ValueError'], False),
-            ('def test_only_rerun(): raise AssertionError("ERR")', ['AssertionError ValueError'], False),
-            ('def test_only_rerun(): raise AssertionError("ERR")', ['AssertionError', 'ValueError'], True),
-        ]
+    "file_text, only_rerun_texts, should_rerun",
```

Comment:
The following lines look not so easy to read after black. I wonder, why we actually have the `file_text` in the parameter list here. We should refactor that out afterwards.

Reply:
I agree, I'll make a follow up issue to refactor it

Reply:
I added this to the PR description @sallner 
When this PR merges I'll raise the 3 issues in the description

Reply:
That sounds good.